### ISSUE TITLE
Markdown preview no longer shows images with path relative to project root

### ISSUE THREAD

**Issue by aaronlindsey** [microsoft/vscode#103343]
I'm noticing the same behavior as in #84728: Images using the syntax `![](/assets/2019-10-24_07_56_26.png)` (where `assets` is a folder at the root of the project) are not displaying in Markdown preview. I tried downgrading to 1.41.1 and did not see the issue. I do see the issue on 1.47.2 and insiders. I see the issue on both my macOS and Ubuntu machines.

- VSCode Version: 1.47.2
- OS Version: Ubuntu 20.04, macOS Catalina

Steps to Reproduce:

1. Exact same steps as in #84728

Does this issue occur when all extensions are disabled?: Yes

**Comment by Daninator1** [comment]
I can reproduce this bug on Windows Server 2016 (Version 1607) as well. On Windows 10 (Version 2004) however, the image is shown correctly. Both operating systems are running VSCode 1.47.3

**Comment by JohnTheFarmer** [comment]
seems solved in 1.47.3  Ubuntu 20.04

**Comment by ardianta** [comment]
@JohnTheFarmer not yet. I use 1.47.3 on Ubuntu 20.04.

![html-netlify md - petanikode - Visual Studio Code_221](https://user-images.githubusercontent.com/4420029/89234685-143b9200-d61f-11ea-909d-d37d9986acd1.png)

```txt
Version: 1.47.3
Commit: 91899dcef7b8110878ea59626991a18c8a6a1b3e
Date: 2020-07-23T15:51:39.791Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Linux x64 5.4.0-42-generic
```

**Comment by JohnTheFarmer** [comment]
@ardianta I believe you forgot to put .. in the image path , like this " ../image/image.png"

**Comment by ardianta** [comment]
Yes, It worked by using `..`, but not relative to root directory.

My directory structure is like this:

```txt
📁 root-dir/
├── 📁 images/
│   └── 🖼️ example.jpg
└── 📁 content/
    └── 📄 my-post.md
```

It would work by using `../images/example.jpg`

```markdown
![](../images/example.jpg)
```

**Comment by JohnTheFarmer** [comment]
@ardianta I tested with your structure , and its working for ![]( /images/example.jpg) without ..
The only difference is i used PNG image and you used JPG.  I tried changed extension to JPG, no longer works. Maybe markdown preview doesn't support JPG or bug in vscode.

**Comment by JohnTheFarmer** [comment]
@ardianta  yeah. its not working , i just realized my previous test has markdown enhanced installed.

**Comment by andrew-maust** [comment]
@ardianta Can you clarify something for me?

Is the expected behavior in this structure
```
📁 root-dir/
├── 📁 images/
│   └── 🖼️ example.jpg
└── 📁 content/
    └── 📄 my-post.md
```
have the path be "../images/example.jpg

and in this case
```
📁 root-dir/
├──📄 my-post.md
├── 📁 images/
│   └── 🖼️ example.jpg
 
```
to have the path be "/images/example.png"?

**Comment by ardianta** [comment]
Hi @andrew-maust thank you for ask me.

I would like to use this structure:

```
📁 root-dir/
├── 📁 images/
│   └── 🖼️ example.jpg
└── 📁 content/
    └── 📄 my-post.md
```

and the path be `/images/example.jpg`.

Or may be we can add configuration to define the root path for images.

**Comment by andrew-maust** [comment]
Is that what the actual intended behavior is or is that personal preference? @mjbvz Could you possibly way in with some input on which case should be the intended behavior?

**Comment by EsbenLorenzen** [comment]
I can reproduce this with Visual Studio Code 1.48.0 on Windows 10.0.1832

I came across this when cloning a Wiki from our Azure DevOps repos.

The structure which gets cloned from the repo:
```
📁 root-dir/
├── 📁 .attachments/
│   └── 🖼️ example.png
└── 📁 content/
    └── 📄 my-post.md
```

Line in MD file that should place image:
```
![example.png](../.attachments/example.png)
```

If i Ctrl+Click on the path in the editor, the image opens correctly.

If I place .attachments folder inside content folder, the Preview window renders the image correctly.

Any suggestions on how to fix this? 
Alternatively reccommend an extension as a workaround, until a fix is released?

**Comment by andrew-maust** [comment]
@mjbvz @ardianta @EsbenLorenzen I do actually have a fix if the desired result is to be able to use "/" from the root directory of the md file is in the root directory. Besides that I don't exactly know where to look to fix it cause I believe if that isn't the fix the issue lies somewhere in the actually render opposed to the link normalizer where the current fix is located.

**Comment by EsbenLorenzen** [comment]
@andrew-maust The result I am looking for is as you described in you previous [comment](https://github.com/microsoft/vscode/issues/103343#issuecomment-670649051).

The notion that the issue might lie in the actual render seems to fit my experience, given that I am able to Ctrl+Click the link in the *.md file, and VSCode opens the correct path and displays the image.

I am not too familiar with the development/release-cycle of VSCode, nor am I familiar with the code base, so I can only be of assistance with testing a possible fix or providing more details of my particular issue.

Let me know if I can be of any help.

**Comment by AndrewKvalheim** [comment]
Per the webview developer tools, the preview of e.g. `![](/images/example.jpg)` fails with status 401 Unauthorized from

```
vscode-webview-resource://x-x-x-x-x/home/user/root-dir/images/example.jpg
```

whereas the preview of `![](../images/example.jpg)` succeeds from

```
vscode-webview-resource://x-x-x-x-x/file///home/user/root-dir/images/example.jpg
```

the only difference being the additional `file///`.

**Comment by platinhom** [comment]
Still the same problem in 1.51.  Could the `markdown.preview` adding a new config for the root directory?

**Comment by rajbos** [comment]
Just noticed this regression in 1.56.1. I've been using the relative path without dots forever, and all of a sudden the images are no longer displayed 😢. Adding two leading dots (../) works in the preview, will need to test if GitHub Pages where I publish to still works with that.  Weird to see this happen. 

#### Do note: this is on a new laptop, will check the old one's setting to see if I have something weird in settings on it.

**Comment by rajbos** [comment]
> Just noticed this regression in 1.56.1. I've been using the relative path without dots forever, and all of a sudden the images are no longer displayed 😢. Adding two leading dots (../) works in the preview, will need to test if GitHub Pages where I publish to still works with that. Weird to see this happen.
> 
> #### Do note: this is on a new laptop, will check the old one's setting to see if I have something weird in settings on it.

Update: old laptop has 1.55.2 where the relative path works without the dots. Found no additional settings.

If I use the dots, they will also get rendered in GitHub Pages, making the links invalid so the images don't display there 🙁 .

**Comment by mjbvz** [comment]
I believe this was fixed as part of #124027

If you still see this issue in the next insiders builds, please share information about your workspace setup so I can investigate
